ui/src/utils/groupByTimeSeriesTransform.js

```diff
@@ -2,50 +2,60 @@ import _ from 'lodash'
 import {shiftDate} from 'shared/query/helpers'
 import {map, reduce, forEach, concat, clone} from 'fast.js'
 
-const groupByMap = (results, responseIndex, groupByColumns) => {
+const flattenGroupBySeries = (results, responseIndex, tags) => {
   if (_.isEmpty(results)) {
     return []
   }
+
+  const tagsKeys = _.keys(tags)
+
+  const seriesArray = _.get(results, [0, 'series'])
+  let valuesAcc
+  valuesAcc = []
+
+  seriesArray.forEach(s => {
+    const tagsToAdd = tagsKeys.map(tk => s.tags[tk])
+    const newValues = s.values.map(v => [v[0], ...tagsToAdd, ...v.slice(1)])
+    valuesAcc = [...valuesAcc, ...newValues]
+  })
   const firstColumns = _.get(results, [0, 'series', 0, 'columns'])
-  const accum = [
+
+  const flattenedSeries = [
     {
-      responseIndex,
       series: [
         {
-          columns: [
-            firstColumns[0],
-            ...groupByColumns,
-            ...firstColumns.slice(1),
-          ],
-          groupByColumns,
+          columns: firstColumns,
+          tagsKeys,
+          isGroupBy: true,
+          tags: _.get(results, [0, 'series', 0, 'tags'], {}),
           name: _.get(results, [0, 'series', 0, 'name'], ''),
-          values: [],
+          values: [...valuesAcc],
         },
       ],
+      responseIndex,
     },
   ]
 
-  const seriesArray = _.get(results, [0, 'series'])
-  seriesArray.forEach(s => {
-    const prevValues = accum[0].series[0].values
-    const tagsToAdd = groupByColumns.map(gb => s.tags[gb])
-    const newValues = s.values.map(v => [v[0], ...tagsToAdd, ...v.slice(1)])
-    accum[0].series[0].values = [...prevValues, ...newValues]
-  })
-  return accum
+  return flattenedSeries
 }
 
-const constructResults = (raw, groupBys) => {
+const constructResults = (raw, isTable) => {
   return _.flatten(
     map(raw, (response, index) => {
       const results = _.get(response, 'response.results', [])
 
       const successfulResults = _.filter(results, r => _.isNil(r.error))
 
-      if (groupBys[index]) {
-        return groupByMap(successfulResults, index, groupBys[index])
+      const tagsFromResults = _.get(results, ['0', 'series', '0', 'tags'], {})
+
+      if (isTable && !_.isEmpty(tagsFromResults)) {
```

Comment:
Consider adding a comment here about how tags are only populated when there's a groupby. Or maybe just turning 
`isTable && !_.isEmpty(tagsFromResults)` into a variable isGroupby/hasGroupby or something like that so it makes sense as to why you call flattenGroupBySeries below.